### PR TITLE
More refactoring PDHCE and preparing for joint detection

### PR DESCRIPTION
**Pull request by davidbenjamin**
@jamesemery here's another one for you.

### CODE REVIEW

**Comment by davidbenjamin** [comment]
@jamesemery Looks like I got that "glorious commit" to work after all.  Now the second-to-last commit will take some scrutiny; the rest are straightforward.

**Comment by jamesemery** [comment]
@davidbenjamin see #8467 where i guess i reviewed this already? Is this substantially different?

**Comment by davidbenjamin** [comment]
@jamesemery Okay, we'ce got some really good tests for EventGroup clustering _and_ branching now.  It's ready for you to take another look.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/PartiallyDeterminedHaplotypeComputationEngine.java
Comment (jamesemery) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/PartiallyDeterminedHaplotypeComputationEngine.java:721]: consider overlapSet -> intersectVsEventGroup to make it a little more legible what is happening here.

Comment (jamesemery) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/PartiallyDeterminedHaplotypeComputationEngine.java:721]: or perhaps just a comment

**Review by jamesemery — APPROVED** [review]
One minor oversight in one of the added tests and then some nothing comments. Feel free to merge this when you are satisfied that you've responded to everything. I would recommend that you go through the exercise of testing this in the FE framework before merging it just to be really sure that there is nothing that is going to break on the full dataset.

**Comment by davidbenjamin** [comment]
Functional equivalence results are unchanged, as they should be.  I will merge now.

**Pull request merged**
